lib/redis_mutex/application.ex

```diff
@@ -17,7 +17,25 @@ defmodule RedisMutex.Application do
 
   def children(_env) do
     redis_url = Application.get_env(:redis_mutex, :redis_url)
+    redix_config = Application.get_env(:redis_mutex, :redix_config)
 
-    [{RedisMutex.Connection, [:redis_mutex_connection, redis_url]}]
+    opts = build_opts(redis_url, redix_config)
+
+    [{RedisMutex.Connection, opts}]
   end
+
+  defp build_opts(nil, nil),
+    do:
+      raise(RedisMutex.Error,
+        message: ":redis_mutex config missing: must specify :redis_url or :redix_config"
+      )
+
+  defp build_opts(url, nil) when is_binary(url), do: [:redis_mutex_connection, url]
+  defp build_opts(nil, opts) when is_list(opts), do: [:redis_mutex_connection, opts]
+
+  defp build_opts(_url, _opts),
+    do:
+      raise(RedisMutex.Error,
+        message: ":redis_mutex config must be either :redis_url or :redix_config, not both"
+      )
```

Comment:
Nit: multiline function formatting

```suggestion
  defp build_opts(_url, _opts) do
    raise RedisMutex.Error,
      message: ":redis_mutex config must be either :redis_url or :redix_config, not both"
  end
```